app/src/main/scala/org/alephium/explorer/config/ExplorerConfig.scala

```diff
@@ -0,0 +1,159 @@
+
+import akka.http.scaladsl.model.Uri
+
+import org.alephium.api.model.ApiKey
+import org.alephium.explorer.error.ExplorerError._
+import org.alephium.protocol.model.NetworkId
+
+object ExplorerConfig {
+
+  /**
+    * Validates raw configurations read from `application.conf` via [[ApplicationConfig]]
+    * and converts it to [[ExplorerConfig]].
+    *
+    * Validates all [[ApplicationConfig]] values before creating [[ExplorerConfig]]
+    * which ensures that Explorer does not get started with invalid configurations.
+    *
+    * @param config Raw `application.conf` properties and values.
+    *
+    * @return A validated [[ExplorerConfig]].
+    */
+  @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
+  def apply(config: ApplicationConfig): Try[ExplorerConfig] =
+    for {
+      groupNum                        <- validateGroupNum(config.blockFlowGroupNum)
+      directCliqueAccess              <- Success(config.blockFlowDirectCliqueAccess)
+      port                            <- validatePort(config.explorerPort)
+      host                            <- validateHost(config.blockFlowHost)
+      readOnly                        <- Success(config.explorerReadOnly)
+      blockFlowUri                    <- validateUri(config.blockFlowHost, config.blockFlowPort)
+      networkId                       <- validateNetworkId(config.blockFlowNetworkId)
+      blockFlowApiKey                 <- validateApiKey(config.blockFlowApiKey)
+      syncPeriod                      <- validateSyncPeriod(config.explorerSyncPeriod)
+      tokenSupplyServiceSyncPeriod    <- validateSyncPeriod(config.tokenSupplyServiceSyncPeriod)
+      hashRateServiceSyncPeriod       <- validateSyncPeriod(config.hashRateServiceSyncPeriod)
+      finalizerServiceSyncPeriod      <- validateSyncPeriod(config.finalizerServiceSyncPeriod)
+      transactionHistoryServicePeriod <- validateSyncPeriod(config.transactionHistoryServicePeriod)
+    } yield
+      ExplorerConfig(
+        groupNum                        = groupNum,
+        directCliqueAccess              = directCliqueAccess,
+        blockFlowUri                    = blockFlowUri,
+        networkId                       = networkId,
+        maybeBlockFlowApiKey            = blockFlowApiKey,
+        host                            = host,
+        port                            = port,
+        readOnly                        = readOnly,
+        syncPeriod                      = syncPeriod,
+        tokenSupplyServiceSyncPeriod    = tokenSupplyServiceSyncPeriod,
+        hashRateServiceSyncPeriod       = hashRateServiceSyncPeriod,
+        finalizerServiceSyncPeriod      = finalizerServiceSyncPeriod,
+        transactionHistoryServicePeriod = transactionHistoryServicePeriod
+      )
+
+  def validateGroupNum(groupNum: Int): Try[Int] =
+    if (groupNum < 0) {
+      Failure(InvalidGroupNumber(groupNum))
+    } else {
+      //Is 0 a valid groupNum? Is there a max limit?
```

Comment:
yes 0 is a valid group. We could check latter what we use to validate group in the node.